src/forms.tsx

```diff
@@ -78,33 +91,97 @@ interface SelectProps {
-  const pickerProps = ({ onChange, value }: any) => ({
-    onValueChange: (v: ReactText, index: number) => onChange(v),
-    selectedValue: value,
-    style: textInput,
-  });
-  return (
-    <>
-    <Controller
-      control={control}
-      render={(params) => <Picker {...pickerProps(params)}>{items}</Picker>}
-      name={name}
-      defaultValue={defaultValue}
-    />
-  </>
-  )
+
+  /** The picker component */
+  const picker = ({ onChange, value }: ControlData<ReactText>) => (
+    <Picker style={textInput} onValueChange={onChange} selectedValue={value}>
+      {items}
+    </Picker>
+  );
+
+  return <Controller
+    name={name}
+    defaultValue={defaultValue}
+    control={control}
+    render={picker}
+  />
 }
 
 ////////////
 // SUBMIT //
 ////////////
 
-export const Submit: FunctionComponent<any> = ({ children, submit, ariaLabel }) => {
-  const title = typeof children === 'string' ? children : 'Submit';
-  return <Button title={title} onPress={submit} accessibilityLabel={ariaLabel} />
+export const Submit: FunctionComponent<any> = ({ children, submit }) => <Btn onPress={submit}>{children}</Btn>;
+
+///////////
+// LABEL //
+///////////
+
+export const Label: FunctionComponent<any> = ({ children }) => {
+  const { label } = useTypography();
+  return <Text style={label}>{children}</Text>
+}
+
+////////////////
+// IMG PICKER //
+////////////////
+
+interface ImgPickerProps {
+  name: string;
+  defaultValue?: string;
+  control?: UseFormMethods['control'];
+  errors?: UseFormMethods['errors'];
+}
+
+export const ImgPicker: FunctionComponent<ImgPickerProps> = ({ name, control, defaultValue }) => {
+
+  /** Assert that the user have given his permission to use the Camera roll */
+  const assertPermission = async () => {
+    const { status } = await ImagePicker.requestCameraRollPermissionsAsync();
+    if (status !== 'granted') {
+      throw new Error('Sorry, we need camera roll permissions to make this work!');
+    }
+  }
+
+  /** Pick the image & trigger onChange on success */
+  const pickImg = async (onChange: (uri: string) => void) => {
+    try {
+      await assertPermission();
+      const result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        allowsEditing: true,
+        aspect: [1, 1],
+        quality: 1,
+      });
+      // If success, update the form value
+      if (!result.cancelled) {
+        onChange(result.uri)
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  /** The image picker component */
+  const imgPicker = ({ onChange, value }: ControlData<string>) => (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+      <Image source={{ uri: value }} style={{ width: 200, height: 200 }} />
+      <Btn onPress={() => pickImg(onChange)}>Pick an image</Btn>
+    </View>
+  );
+
+  /** Controller wrapper around the image picker to use it inside a form */
+  return <Controller
```

Comment:
Why do we need a `controlled component` here, why not keeping it `uncontrolled` ?

Reply:
I don't know what `uncontrolled` is. But `Controller` wrapper is here to put the result of the img picker directly in the form. Like that you can add validation, update on submit, ...
I think the is the cleanest way to deal with forms that are more than 2 fields long. Happy to explain further if needed.

Reply:
Hmm 🤔 
In the [doc](https://react-hook-form.com/faqs/) they have examples of forms with more than 2 fields long that are not wrapped into a `<Controller>`  ([this one for eg](https://codesandbox.io/s/rwpz2jn6ln)), that's why I'm confused.

Reply:
We could try, but I don't see how to connect the img picker value with this method. I think this works only because react-hooks-form has been build for native html form tags (input, select, textarea, ....). If you have something a little been custom you need to use a wrapper to connect your custom component & the form. I havent been too deep into that, but this is what I understood from the doc.